Modules/JavaScript/Asynchronous/task2.js

```diff
@@ -0,0 +1,41 @@
+console.log(`
+  Task #2
+
+    Write ReversePromise class so that ‘then’ functions are calling from the end to the start. 
+
+      Example:
+
+      let promise = new ReversePromise((resolve) => {
+        console.log(1);
+        resolve();
+      })
+      .then(() => console.log(2))
+      .then(() => console.log(3))
+      .then(() => console.log(4))
+      //1, 4, 3, 2
+`);
+
+class ReversePromise extends Promise {
+  list = [];
+
+  then(b) {
+    this.list.push(b);
+
+    setTimeout(() => {
```

Comment:
```suggestion
    setTimeout(() => this.list.pop()())
```
Interesting solution, my only suggestion would be to simplify it like this. The zero is not needed as well, as it is the default value